### ISSUE TITLE
No error / warning on port conflict

### ISSUE THREAD

**Issue by alesj** [quarkusio/quarkus#3370]
I had some really old process (some GAE server) running on 8081, forgot about it, but when running Quarkus REST quickstart, the tests didn't pass -- as they run on 8081 ... they just failed with wrong test result.
Imho, Quarkus should barf as there was a port conflict,
where actually it was a misleading error -- plain 404 instead of expected 200 http response code.

**Comment by rsvoboda** [comment]
I played with rest-json quicstart, having 8081 occupied by `python -m SimpleHTTPServer 8081`

Server boots well:
```
java -Dquarkus.http.port=8081 -jar target/rest-json-1.0-SNAPSHOT-runner.jar
2019-08-07 15:49:40,844 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 1.116s. Listening on: http://[::]:8081
2019-08-07 15:49:40,855 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jsonb]
```

I can access properly (only) `http://[::]:8081/fruits`.
`http://127.0.0.1:8081/fruits` and `http://MY_LOCAL_IP:8081/fruits` return 404 from SimpleHTTPServer.

With `java -Djava.net.preferIPv4Stack=true -Dquarkus.http.port=8081 -jar target/rest-json-1.0-SNAPSHOT-runner.jar` I receive java.net.BindException: Address already in use. Java process is stuck and doesn't end :)

**Comment by machi1990** [comment]
> With `java -Djava.net.preferIPv4Stack=true -Dquarkus.http.port=8081 -jar target/rest-json-1.0-SNAPSHOT-runner.jar` I receive java.net.BindException: Address already in use. Java process is stuck and doesn't end :)

@rsvoboda I had similar results.

**Comment by alesj** [comment]
Again bitten by this ... :-)

**Comment by rsvoboda** [comment]
@alesj two more times and you will fix it on your own ;)

**Comment by stuartwdouglas** [comment]
This is fixed in 1.3.0. The first test will fail with a bind exception, the rest will be skipped.

**Comment by rsvoboda** [comment]
I tried running tests and running the app but I didn't get bind exception.

On macOS, port occupied by python SimpleHTTPServer
```
python -m SimpleHTTPServer 8081
Serving HTTP on 0.0.0.0 port 8081 ...
```

Using rest-json-quickstart, executing `mvn clean verify -Dquarkus.platform.version=1.3.0.Final`
```
[INFO] Running org.acme.rest.json.LegumeResourceTest
2020-03-19 08:30:00,985 INFO  [io.quarkus] (main) Quarkus 1.3.0.Final started in 1.622s. Listening on: http://0.0.0.0:8081
2020-03-19 08:30:00,987 INFO  [io.quarkus] (main) Profile test activated.
2020-03-19 08:30:00,987 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jsonb]
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.422 s <<< FAILURE! - in org.acme.rest.json.LegumeResourceTest
[ERROR] testList  Time elapsed: 0.948 s  <<< FAILURE!
java.lang.AssertionError:
1 expectation failed.
Expected status code <200> but was <404>.

	at org.acme.rest.json.LegumeResourceTest.testList(LegumeResourceTest.java:19)
```
SimpleHTTPServer prints these logs
```
127.0.0.1 - - [19/Mar/2020 08:33:37] code 404, message File not found
127.0.0.1 - - [19/Mar/2020 08:33:37] "GET /legumes HTTP/1.1" 404 -
```
